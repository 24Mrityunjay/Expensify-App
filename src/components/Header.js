import React from 'react';
import { NavLink} from 'react-router-dom';

const Header = ()=>(
    <header>
       <h1> Expense App</h1> 
       <NavLink to='/' activeClassName='is-active' exact={true }>Go Home</NavLink><br/>
       <NavLink to='/create' activeClassName='is-active'>Go To Create Page</NavLink><br/>
       <NavLink to='/edit' activeClassName='is-active'>Go To Edit Page</NavLink><br/>  
    </header>
);

export default Header;