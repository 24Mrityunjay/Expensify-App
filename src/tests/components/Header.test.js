import React from 'react';
import { shallow } from 'enzyme';
import ReactShallowRenderer  from "react-test-renderer/shallow";
//import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

test('Should render header correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

// test('Should render header correctly', () => {
//    const wrapper = shallow(<Header/>);
//    expect(wrapper.find('h1').length).toBe(1);
// });
// test('Should render header correctly', () => {
//    const wrapper = shallow(<Header />);
//    expect(wrapper).toMatchSnapshot();
// });