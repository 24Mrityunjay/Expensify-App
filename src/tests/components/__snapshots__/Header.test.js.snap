// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header>
  <h1>
     Expense App
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Go Home
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Go To Create Page
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    to="/edit"
  >
    Go To Edit Page
  </NavLink>
  <br />
</header>
`;
